package Game;

message MessageType
{
	enum EMessageType
	{
		PING = 1;
		PONG = 2;
		REMOTE_DISCONNECTION = 3;
		MOVE = 4;
		NEXT_TURN = 5;
		ATTACK = 6;
	}
	required EMessageType type = 1;
	
	optional Ping ping = 2;
	optional Pong pong = 3;
	optional Move move = 4;
	optional Attack attack = 5;
}

message Ping
{
	required uint32 id = 1;
}

message Pong
{
	required uint32 id = 1;
}

message Case
{
	required uint32 x = 1;
	required uint32 y = 2;
}

message Move
{
	required Case from = 1;
	required Case to = 2;
}

message Attack
{
	required Case from = 1;
	required Case target = 2;
}

message CaseMap
{
	enum ECaseType
	{
		GRASS = 1;
		ROAD = 2;
	}
	
	enum EUnitType
	{
		RED_GUERRIER = 1;
		BLUE_GERRIER = 2;
	}
	
	required ECaseType type = 1;
	optional EUnitType unit = 2;
}

message Map
{
	required string name = 1;
	repeated CaseMap cases = 2;
}